packages/contracts-core/contracts/scripts/Config.sol

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import {UpgradeBeaconProxy} from "../upgrade/UpgradeBeaconProxy.sol";
+import {UpgradeBeacon} from "../upgrade/UpgradeBeacon.sol";
+import {UpgradeBeaconController} from "../upgrade/UpgradeBeaconController.sol";
+import {UpdaterManager} from "../UpdaterManager.sol";
+import {XAppConnectionManager} from "../XAppConnectionManager.sol";
+
+import "forge-std/Vm.sol";
+import "forge-std/StdJson.sol";
+import "forge-std/Test.sol";
+
+abstract contract Config {
+    using stdJson for string;
+
+    Vm private constant vm =
+        Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    string internal config;
+
+    struct Upgrade {
+        UpgradeBeacon beacon;
+        address implementation;
+        UpgradeBeaconProxy proxy;
+    }
+
+    modifier onlyInitialized() {
+        require(isInitialized(), "not initialized");
+        _;
+    }
+
+    function __Config_initialize(string memory file) internal {
+        require(!isInitialized(), "already init");
+        config = vm.readFile(file);
+    }
+
+    function isInitialized() public view returns (bool) {
+        return bytes(config).length != 0;
+    }
+
+    function core(string memory domain) private returns (string memory) {
```

Comment:
prob cleaner to do `config.readString(path)` and use the stdJson helpers?

If you agree, we should do it everywhere.

Reply:
I don't think that does the same thing. that reads a string from the json value, while this returns the entire object.

that said, this method is redundant and unused, and ought to be removed